cmd/traffic/cmd/manager/config/config.go

```diff
@@ -115,9 +124,27 @@ func (c *config) refreshFile(ctx context.Context, data map[string]string) {
 		c.clientYAML = nil
 		dlog.Debugf(ctx, "Cleared client config")
 	}
+
+	c.agentEnv = AgentEnv{}
+	if yml, ok := data[agentEnvConfigFileName]; ok {
+		data, err := yaml.YAMLToJSON([]byte(yml))
+		if err == nil {
+			err = json.Unmarshal(data, &c.agentEnv)
+		}
```

Comment:
```
nit - don't need to address it now, but at some point. YAML 1.2 much compliant with JSON assuming JSON or core schemas are used (core is the recommended default for processors). There are a few differences that should be irrelevant in this context.
As such, one can do
```suggestion
		data, err := yaml.Unmarshal([]byte(yml))
```
to handle both file types.

Reply:
There's no yaml that can handle the JSON functionality in the json package that we're using now. (such as the `json` tags and `omitzero` See https://github.com/go-json-experiment/json for more info.